File: bugbank-selenium/src/test/java/bugbank/pages/LoginPage.java
package bugbank.pages;

import com.beust.ah.A;
import org.openqa.selenium.By;
import org.openqa.selenium.JavascriptExecutor;
import org.openqa.selenium.Keys;
import org.openqa.selenium.support.ui.ExpectedConditions;

public class LoginPage extends BasePage {

    //region ELEMENTOS
    private static final By campoEmail = By.cssSelector("input[type='email']");
    private static final By campoSenha = By.cssSelector("input[type='password']");
    private static final By campoSenhaMensagem = By.cssSelector("div.card__login div.login__password [class*='input__child'] p");
    private static final By campoEmailMensagem = By.cssSelector("div.card__login > form > div.input__child > p");
    private static final By botaoAcessar = By.cssSelector("div.login__buttons > button[type=submit]");
    private static final By botaoRegistrar = By.cssSelector("div.login__buttons > button[type=button]");
    private static final By textoModal = By.id("modalText");
    private static final By botaoFecharModal = By.id("btnCloseModal");
    //endregion

    //region FLUXOS
    public static void logar(String email, String senha) {
        acessarPagina();
        preencherCampoEmail(email);
        preencherCampoSenha(senha);
        clicarBotaoAcessar();
    }

    public static void acessarPagina() {
        driver.get("https://bugbank.netlify.app");
    }
    //endregion

    //region MÉTODOS
    public static void preencherCampoEmail(String email) {
        sendKeys(campoEmail, email);
    }

    public static void preencherCampoSenha(String senha) {
        sendKeys(campoSenha, senha);
        sendKeys(campoSenha, Keys.TAB);
    }

    public static void clicarBotaoAcessar() {
        click(botaoAcessar);
    }

    public static void clicarBotaoRegistrar() {
        click(botaoRegistrar);
    }

    public static String getMensagemCampoEmail() {
        return getText(campoEmailMensagem);
    }

    public static String getMensagemCampoSenha() {
        return getText(campoSenhaMensagem);
    }

    public static String getTextoModal() {
        wait.until(ExpectedConditions.visibilityOfElementLocated(textoModal));
        return getText(textoModal);
    }
    //endregion
}

File: bugbank-cypress/cypress/e2e/pages/login-page.js
export class LoginPage {

    static url = 'https://bugbank.netlify.app'
    static emailInput = "div.card__login input[type='email']"
    static senhaInput = "div.card__login input[type='password']"
    static botaoAcessar = "div.card__login > form > div.login__buttons > button[type=submit]"
    static botaoRegistrar = "div.card__login > form > div.login__buttons > button[type=button]"
    static mensagemCampoEmail = "div.card__login > form > div.input__child > p"
    static mensagemCampoSenha = "div.card__login div.login__password [class*='input__child'] p"
    static textoModal = "#modalText"
    
    static logar(email, senha) {
        this.preencherCampoEmail(email);
        this.preencherCampoSenha(senha);
        this.clicarBotaoAcessar();
    }

    static acessarPagina() {
        cy.visit(this.url);
    }

    static preencherCampoEmail(email) {
        cy.get(this.emailInput).type(email);
    }

    static preencherCampoSenha(senha) {
        cy.get(this.senhaInput).type(senha).tab();
    }

    static clicarBotaoAcessar() {
        cy.get(this.botaoAcessar).click();
    }

    static clicarBotaoRegistrar() {
        cy.get(this.botaoRegistrar).click();
    }

    static getMensagemCampoEmail() {
        return cy.get(this.mensagemCampoEmail).invoke('text');
    }

    static getMensagemCampoSenha() {
        return cy.get(this.mensagemCampoSenha).invoke('text');
    }

    static getTextoModal() {
        cy.get(this.textoModal).should('be.visible');
        return cy.get(this.textoModal).invoke('text');
    }
}